File: claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.Attributes;
import se.curity.identityserver.sdk.attribute.AuthenticationAttributes;
import se.curity.identityserver.sdk.claims.ClaimsProvider;
import se.curity.identityserver.sdk.claims.RequestedClaimAttribute;
import se.curity.identityserver.sdk.data.authorization.ScopeValue;
import se.curity.identityserver.sdk.oauth.OAuthClient;

import java.util.Set;

public final class ${pluginName}ClaimsProvider implements ClaimsProvider
{
    private final ${pluginName}ClaimsProviderConfig _configuration;

    public ${pluginName}ClaimsProvider(${pluginName}ClaimsProviderConfig configuration)
    {
        _configuration = configuration;

        // See examples of Claims Providers on GitHub: https://github.com/search?q=topic%3Aclaims-provider+org%3Acurityio
    }

    @Override
    public Attributes getClaimValues(Set<RequestedClaimAttribute> requestedClaimAttributes,
                                     Set<ScopeValue> scopeValues,
                                     AuthenticationAttributes userAuthenticationAttributes,
                                     OAuthClient client)
    {
        throw new UnsupportedOperationException();
    }
}

File: kotlin-claims-provider-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__ClaimsProvider.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.attribute.Attributes
import se.curity.identityserver.sdk.attribute.AuthenticationAttributes
import se.curity.identityserver.sdk.claims.ClaimsProvider
import se.curity.identityserver.sdk.claims.RequestedClaimAttribute
import se.curity.identityserver.sdk.data.authorization.ScopeValue
import se.curity.identityserver.sdk.oauth.OAuthClient

class ${pluginName}ClaimsProvider(private val _configuration: ${pluginName}ClaimsProviderConfig): ClaimsProvider
{
    // See examples of Claims Providers on GitHub: https://github.com/search?q=topic%3Aclaims-provider+org%3Acurityio

    override fun getClaimValues(requestedClaimAttributes: Set<RequestedClaimAttribute>,
                                scopeValues: Set<ScopeValue>,
                                userAuthenticationAttributes: AuthenticationAttributes,
                                client: OAuthClient): Attributes
    {
        TODO()
    }
}
